### PR TITLE
Adding stronger types to Cotter interfaces

### PR DESCRIPTION
**Pull request by theFong**

### CODE REVIEW

**Comment by theFong** [comment]
I would like to add stronger types to the `verificationProccessPromise` but looks like an interface for the verifySuccess payload has not been defined yet?

**Comment by theFong** [comment]
I added a payload interface to type `verificationProccessPromise` resolution. Probably could be used in other publicly exposed parts of the api. Let me know if you would like the naming changed etc or want it moved somewhere else :)

File: src/API.ts [hunk 1]
```diff
@@ -138,7 +139,7 @@ class API {
     identifier: string,
     credential: any,
     origin: string
-  ): Promise<any> {
+  ): Promise<Payload> {
```
Comment (theFong) [inline src/API.ts]: Assuming this is the same interface

File: src/API.ts [hunk 2]
```diff
@@ -211,7 +212,7 @@ class API {
     credential: any,
     origin: string,
     publicKey: string
-  ): Promise<any> {
+  ): Promise<Payload> {
```
Comment (theFong) [inline src/API.ts]: Assuming this is the same interface

**Comment by putrikarunia** [comment]
Hey, thank you for contributing! I think verifySuccess is supposed to be the response that is returned to the client because it was used for `resolve(self.verifySuccess)`. Is it different when you tried it? 

It should be like this:

```typescript
export interface VerifySuccess {
  email?: String;
  phone?: String;
  oauth_token: OAuthToken;
  user: User;
  identifier?: String; // only from WebAuthn same as email/phone
  status?: String; // only from WebAuthn = SUCCESS or CANCELED
}
```
where `OAuthToken` is from `handler/TokenHandler.ts` but missing some parts:
```diff

export interface OAuthToken {
  access_token: string;
  id_token: string;
  token_type: string;
  refresh_token: string;
+  expires_in: number;
+  auth_method: string;
}
```
and `User` from `models/User`.

**Comment by theFong** [comment]
> Hey, thank you for contributing! I think verifySuccess is supposed to be the response that is returned to the client because it was used for `resolve(self.verifySuccess)`. Is it different when you tried it?
> 
> It should be like this:
> 
> ```ts
> export interface VerifySuccess {
>   email?: String;
>   phone?: String;
>   oauth_token: OAuthToken;
>   user: User;
>   identifier?: String; // only from WebAuthn same as email/phone
>   status?: String; // only from WebAuthn = SUCCESS or CANCELED
> }
> ```
> 
> where `OAuthToken` is from `handler/TokenHandler.ts` but missing some parts:
> 
> ```diff
> export interface OAuthToken {
>   access_token: string;
>   id_token: string;
>   token_type: string;
>   refresh_token: string;
> +  expires_in: number;
> +  auth_method: string;
> }
> ```
> 
> and `User` from `models/User`.

Oops! You are totally correct the payload interface was something I was logging in the onBegin and not on the verifySuccess. I have added those accordingly.
Added the VerifySuccess interface to binder.ts. Is this the proper file to use?

File: src/binder.ts [hunk 3]
```diff
@@ -76,3 +83,46 @@ export interface Config {
...
+  device_name: string;
+  device_type: string;
+  identifier: string;
+  identifier_type: string;
+  origin: string;
+  public_key: string;
+  redirect_url: string;
+  signature: string;
+  timestamp: string;
+}
```
Comment (putrikarunia) [inline src/binder.ts]: the Payload that is passed into `OnBegin` should be like this
```typescript
export interface Payload {
  auth_required: boolean;
  device_name: string; // User Agent
  device_type: "BROWSER";
  identifier: string; // User's email/phone
  identifier_type: "EMAIL" | "PHONE";
}
```
What you're seeing above (challenge, challenge_id, etc) is probably from some other console log which I should remove from the code.

**Review by putrikarunia — CHANGES_REQUESTED** [review]
Looks good 👍, just need to update the `Payload` to the one I posted in the comment.

**Comment by theFong** [comment]
Changed Payload to reflect real object and changed name from `Payload` to `OnBeginPayload`

**Review by putrikarunia — APPROVED** [review]

**Pull request merged**